src/components/ResizeList/ResizeList.tsx

```diff
@@ -0,0 +1,98 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { join } from '../../util/classNames';
+import { BaseComponentProps } from '../../util/BaseComponent/props';
+import { ResizeGroup } from 'office-ui-fabric-react/lib/ResizeGroup';
+import { GutterSize, LayoutList, LayoutListItem, LayoutListProps } from '../LayoutList';
+
+export { GutterSize };
+
+interface ResizeListData {
+  count: number;
+  cacheKey: string;
+}
+
+export interface ResizeListProps extends BaseComponentProps {
+  /**
+   * A list of child elements to render. Items which don't fit the container width will not be rendered.
+   */
+  items: React.ReactNode[];
+
+  /**
+   * An additional item which will always be rendered at the end of th elist.
+   */
+  actionItem?: React.ReactNode;
+
+  /**
+   * Amount of spacing between list items. Defaults to 4px.
+   * @default GutterSize.XSMALL
+   */
+  gutterSize?: LayoutListProps['gutterSize'];
+}
+
+/**
+ * ResizeList renders a items on a single row, without rendering any items which don't fit in the available width.
+ * An optional "actionItem" can be provided, which will always be rendered after the other visible items.
+ */
+export default class ResizeList extends React.Component<ResizeListProps> {
+  public render() {
+    const data: ResizeListData = {
+      count: this.props.items.length,
+      cacheKey: this.props.items.length.toString(),
```

Comment:
This won't create problems if someone swaps one item with a different one?

Reply:
the cache just prevents recalculation, not a re-render, so everything is fine as long as the items don't change size. it looks like that's why the Fabric component offers a remeasure() public method, i'll surface that and componentRef

Reply:
Ah fair.

Reply:
actually that was a bit of a rabbit hole, `remeasure` only triggers the same container remeasure that window.resize triggers. looks like Fabric's ResizeGroup doesn't expect your items to change dimensions. we don't currently have this use case, but if we do we'll need to update Fabric's component to handle it